tests/common.py

```diff
@@ -344,26 +344,22 @@ def __init__(self, domain=None, dependencies=None, setup=None,
         self.DOMAIN = domain
         self.DEPENDENCIES = dependencies or []
         self.REQUIREMENTS = requirements or []
-        self._setup = setup
 
         if config_schema is not None:
             self.CONFIG_SCHEMA = config_schema
 
         if platform_schema is not None:
             self.PLATFORM_SCHEMA = platform_schema
 
+        if setup is not None:
+            # We run this in executor, wrap it in function
+            self.setup = lambda *args: setup(*args)
+
         if async_setup is not None:
             self.async_setup = async_setup
 
-    def setup(self, hass, config):
-        """Set up the component.
-
-        We always define this mock because MagicMock setups will be seen by the
-        executor as a coroutine, raising an exception.
-        """
-        if self._setup is not None:
-            return self._setup(hass, config)
-        return True
+        if not setup and not async_setup:
```

Comment:
Style suggestion: You're using `is not None` above, but here you're suddenly checking for nullability (if i'm correct) with `__nonzero__ `. [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations): 

> Use is not operator rather than not ... is. While both expressions are functionally identical, the former is more readable and preferred.
> 
> **Yes:**
> 
> ```python
> if foo is not None:
> ```
> 
> **No:**
> 
> ```python
> if not foo is None:
> ```